plugins/module_utils/network/vyos/config/ntp/ntp.py

```diff
@@ -0,0 +1,177 @@
+It is in this file where the current configuration (as dict)
+is compared to the provided configuration (as dict) and the command set
+necessary to bring the current configuration to its desired end-state is
+created.
+"""
+
+
+from copy import deepcopy
+
+from ansible.module_utils.six import iteritems
+from ansible_collections.ansible.netcommon.plugins.module_utils.network.common.utils import (
+    dict_merge,
+)
+from ansible_collections.ansible.netcommon.plugins.module_utils.network.common.rm_base.resource_module import (
+    ResourceModule,
+)
+from ansible_collections.vyos.vyos.plugins.module_utils.network.vyos.facts.facts import (
+    Facts,
+)
+from ansible_collections.vyos.vyos.plugins.module_utils.network.vyos.rm_templates.ntp import (
+    NtpTemplate,
+)
+
+
+class Ntp(ResourceModule):
+    """
+    The vyos_ntp config class
+    """
+
+    def __init__(self, module):
+        super(Ntp, self).__init__(
+            empty_fact_val={},
+            facts_module=Facts(module),
+            module=module,
+            resource="ntp",
+            tmplt=NtpTemplate(),
+        )
+        self.parsers = [
+            "allow_clients",
+            "listen_addresses",
+            "name",
+            "options"
+        ]
+
+    def execute_module(self):
+        """ Execute the module
+
+        :rtype: A dictionary
+        :returns: The result from module execution
+        """
+        if self.state not in ["parsed", "gathered"]:
+            self.generate_commands()
+            self.run_commands()
+        return self.result
+
+    def generate_commands(self):
+        """ Generate configuration commands to send based on
+            want, have and desired state.
+        """
+
+        wantd = self._ntp_list_to_dict(self.want)
+        haved = self._ntp_list_to_dict(self.have)
+        
+
+        # if state is merged, merge want onto have and then compare
+        if self.state == "merged":
+            wantd = dict_merge(haved, wantd)
+
+        # if state is deleted, empty out wantd and set haved to wantd
+        if self.state == "deleted":
+            haved = {
+                k: v for k, v in iteritems(haved) if k in wantd or not wantd
+            }
+            wantd = {}
+            commandlist = []
+            for k,have in iteritems(haved):
+                for k,val in iteritems(have):
+                    if k not in commandlist and k!="options":
+                        commandlist.append(k)
+            for k in commandlist:
+                if k == "allow_clients":
+                    self.commands.append(
+                        "delete system ntp allow-clients"
+                    )
+                elif k=="listen_addresses":
+                     self.commands.append(
+                        "delete system ntp listen-address"
+                    )
+                else:
+                    self.commands.append(
+                        "delete system ntp server"
+                    )
+    
```

Comment:
Instead of using the commands, directly, can we use the rm_template's setval and remval, to form the commands?